src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2040 @@
+        open connection with RemoteEx client.
+        :parameter host: IP-adress or hostname
+        :parameter port: port for sending/receiving commands
+        :parameter port_d: port for reading images
+        return: connection to RemoteEx command and data port
+        """
+        # connect to sockets
+        try:
+            self._commandport = socket.create_connection((self.host, self.port), timeout=5)
+            self._dataport = socket.create_connection((self.host, self.port_d), timeout=5)
+        except socket.timeout as msg:
+            raise model.HwError(msg, "Failed to connect to '%s using port %d'. Check the server "
+                                "is connected to the network, turned "
+                                " on, and correctly configured." % (self.host, self.port))
+        except socket.error as msg:
+            raise model.HwError(msg, "Failed to connect to '%s:%d'. Check ...." % (self.host, self.port))
+
+        # check if connection returns correct response
+        try:
+            message = self._commandport.recv(self.port)
+            if message != 'RemoteEx Ready\r':
+                raise ValueError("Connection to port %s not successfull. "
+                                 "Response %s from server is not as expected." % (self.port, message))
+        except socket.timeout as msg:
+            raise model.HwError(msg, "Failed to receive response from '%s:%d'. Check ..." % (self.host, self.port))
+
+        try:
+            message_d = self._dataport.recv(self.port_d)
+            if message_d != 'RemoteEx Data Ready\r':
+                raise ValueError("Connection to port %s not successfull. "
+                                 "Response %s from server is not as expected." % (self.port_d, message))
+        except socket.timeout as msg:
+            raise model.HwError(msg, "Failed to receive response from '%s:%d'. Check ..." % (self.host, self.port_d))
+
+        # set timeout
+        self._commandport.settimeout(1.0)
+        self._dataport.settimeout(5.0)
+
+        return self._commandport, self._dataport
+
+    def _start_receiverThread(self):
+        """Start the receiver thread, which keeps listening to the response of the command port."""
+        self.t_receiver = threading.Thread(target=self.readCommandResponse)
+        self.t_receiver.start()
+
+    def _closeConnection(self):
+        """
+        Close connection to RemoteEx.
+        """
+        self._commandport.close()
+        self._dataport.close()
+
+    def terminate(self):
+        """
+        Close App (HPDTA) and RemoteEx and close connection to RemoteEx. Called by backend.
+        """
+        # terminate children
+        for child in self.children.value:
+            child.terminate()
+
+        self.should_listen = False  # terminates receiver thread
+        if self.t_receiver.isAlive():
+            self.t_receiver.join(5)
+        self._closeConnection()
+
+    def sendCommand(self, func, timeout=5, *args):
```

Comment:
The addition of "timeout" as second argument broke a lot of calls. As you haven't run the code, you haven't noticed :-/
Unfortunately, Python 2 doesn't allow you to put it at the end. (Python 3 does!).
I'd suggest to write `sendCommand(self, func, *args, **kwargs):`

And in the code, check if kwargs has "timeout". If it has anything else => raise ValueError.

Reply:
I could also write down this way?
```
timeout = kwargs.pop("timeout", 5)
```